src/main/java/org/opensearch/knn/index/SpaceType.java

```diff
@@ -104,18 +116,29 @@ public float scoreTranslation(float rawScore) {
         }
 
         @Override
-        public VectorSimilarityFunction getVectorSimilarityFunction() {
-            return VectorSimilarityFunction.MAXIMUM_INNER_PRODUCT;
+        public KNNVectorSimilarityFunction getKnnVectorSimilarityFunction() {
+            return KNNVectorSimilarityFunction.MAXIMUM_INNER_PRODUCT;
         }
     },
     HAMMING_BIT("hammingbit") {
         @Override
         public float scoreTranslation(float rawScore) {
             return 1 / (1 + rawScore);
         }
+
+        @Override
+        public boolean isSupported(VectorDataType vectorDataType) {
+            return true;
+        }
```

Comment:
why this is always returning true?

Reply:
Yes hamming bit was there before we added the support for fields. But Hamming bit is not supported with Floats. The field was not the knn field. Check this Link: https://opensearch.org/docs/latest/search-plugins/knn/knn-score-script/#getting-started-with-the-score-script-for-binary-data

Reply:
Updated the check